addons/account/models/account_move_line.py

```diff
@@ -3104,6 +3097,17 @@ def _prepare_analytic_distribution_line(self, distribution, account_ids, distrib
             'category': 'invoice' if self.move_id.is_sale_document() else 'vendor_bill' if self.move_id.is_purchase_document() else 'other',
         }
 
+    def _prepare_analytic_distribution_vals(self):
```

Comment:
`analytic_distribution` doesn't really make sense in these values. It also prevents the caching above to work properly...

Would it not make more sense to have a separate function `_related_analytic_distribution` that retrieves the distribution of the SO/PO/...? Then we can merge both distributions here?
You might need to provide the plans already known in the existing distribution to compute `_get_distribution correctly though.`